### PR TITLE
[Anuj] Adding functionality for creating Fabric domains

### LINKED ISSUE #692: Minor enhancements to boostrap script
This issue it to make some minor enhancements to the bootstrap script. It includes the following changes:

- Deriving "resource group" and "fabric capacity" name based on "FABRIC_PROJECT_NAME".
- Adding functionality for "Domain" creation.
- ...

### PR DESCRIPTION
**Pull request by promisinganuj**
# Type of PR

- Documentation changes
- Code changes

## Purpose

- To add new functionality for creating Fabric domains.

## Does this introduce a breaking change? If yes, details on what can break

- No, it's a new addition.

## Author pre-publish checklist

- [x] Added test to prove my fix is effective or new feature works
- [x] No PII in logs
- [x] Made corresponding changes to the documentation

## Validation steps

- Do an E2E deployment.

## Issues Closed or Referenced

- Closes #692 
- References #674

### CODE REVIEW

File: single_tech_samples/fabric/fabric_ci_cd/README.md [hunk 1]
```diff
@@ -32,8 +32,7 @@ This repo contains a code sample for establishing a CI/CD process for Microsoft
 ### Pre-requisites
 
 - Ensure *always* latest Fabric Token is added to the .env file (see instructions below).
-- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) and [jq](https://stedolan.github.io/jq/) are installed.
-- [jq](https://stedolan.github.io/jq/)
+- [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli?view=azure-cli-latest) and [jq](https://jqlang.github.io/jq/download/) are installed.
```
Comment (ydaponte) [inline single_tech_samples/fabric/fabric_ci_cd/README.md:35]:
```suggestion
- [Azure CLI](https://docs.microsoft.com/cli/azure/install-azure-cli?view=azure-cli-latest) and [jq](https://jqlang.github.io/jq/download/) are installed.
```

File: single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg
Comment (ydaponte) [inline single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg:1]: As the script is assigning the Domains to the workspaces, shouldn't we represent the connection with a dotted line? I know that there is not so much space - but would be inline with what is done in the script.

**Review by ydaponte — APPROVED** [review]
Great job! Approving with a couple of suggestions. Tks!

File: single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg
Comment (promisinganuj) [reply single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg:1]: Good point, I ran out of the real-estate in the diagram. Let me try again.

File: single_tech_samples/fabric/fabric_ci_cd/README.md (hunk 1 above)
Comment (promisinganuj) [reply single_tech_samples/fabric/fabric_ci_cd/README.md:35]: Updated

File: single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg
Comment (promisinganuj) [reply single_tech_samples/fabric/fabric_ci_cd/images/fabric-cicd.drawio.svg:1]: Updated
